### PR TITLE
feat(price): begin lowest layer for getting price from a platform

### PR DESCRIPTION
**Pull request by Milerius**

### CODE REVIEW

File: src/price/coinpaprika.price.platform.cpp [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            DVLOG_F(loguru::Verbosity_INFO, "request: %s", final_uri.c_str());
+            auto response = RestClient::get(final_uri);
+            DVLOG_F(loguru::Verbosity_INFO, "response: %s\nstatus: %d", response.body.c_str(), response.code);
+            if (response.code == 200) {
+                auto resp_json = nlohmann::json::parse(response.body);
+                auto price = resp_json["quotes"][currency_pair.quote.value()]["price"].get<double>();
+                return price;
+            }
+        }
+        return 0;
```
Comment (Milerius) [inline src/price/coinpaprika.price.platform.cpp]: There is no proper error handling right now, we need to discuss about it

**Comment by Milerius** [comment]
There is no real error handling for the moment.

File: src/price/coinpaprika.price.platform.hpp [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+{
+    class coinpaprika_price_platform : public abstract_price_platform
+    {
+    public:
+        explicit coinpaprika_price_platform(const config &cfg) : abstract_price_platform(cfg)
+        {
+
+        }
+
+        st_price get_price(antara::pair currency_pair) final;
```
Comment (ca333) [inline src/price/coinpaprika.price.platform.hpp:32]: if this is considered WIP, wouldn't `final` prevent us from overloading `getprice()` to use other parameters?

Comment (Milerius) [reply src/price/coinpaprika.price.platform.hpp:32]: There will be different overload of the function and new signature in the abstract class. So we can set final for every overload. Final mean here that we are at the lowest layer.

**Review by ca333 — APPROVED** [review]

**Pull request merged**